src/libopensc/card-belpic.c

```diff
@@ -231,11 +232,19 @@ static int belpic_init(sc_card_t *card)
 		if(get_carddata(card, carddata, sizeof(carddata)) < 0) {
 			return SC_ERROR_INVALID_CARD;
 		}
-		if (carddata[BELPIC_CARDDATA_OFF_APPLETVERS] >= 0x17) {
-			key_size = 2048;
+		if (carddata[BELPIC_CARDDATA_OFF_APPLETVERS] >= 0x18) {
+			flags = SC_ALGORITHM_ECDSA_HASH_NONE | SC_ALGORITHM_ECDSA_RAW;
+			_sc_card_add_ec_alg(card, 256, flags, 0, NULL);
+			_sc_card_add_ec_alg(card, 384, flags, 0, &(struct sc_object_id){{1, 3, 132, 0, 34, -1}});
```

Comment:
please, do the oids similarly as the PIV -- create global static list of the oids:

https://github.com/OpenSC/OpenSC/blob/8c1a26023c0da489577f253d596ce0a488252faa/src/libopensc/card-piv.c#L548-L549

and then pass them to this function:

https://github.com/OpenSC/OpenSC/blob/8c1a26023c0da489577f253d596ce0a488252faa/src/libopensc/card-piv.c#L5755-L5758

using the cycle might make it more readable too.

Reply:
Most cards do not add the curve to _sc_card_add_ec_alg.  And if you don't have documentation on what the others are, I would suggest you don't add any, since it appears to be a read only card and the curve is not really needed in _sc_card_add_ec_alg, as it is in public key and public key in certificates. 
The curve was added to PIV for the Windows  minidriver when creating keys for ECDH and Windows only supports 3 at this time.   `grep curve minidriver/minidriver.c`

Reply:
Ok, then. So lets not add any OIDs. But just make sure the sizes make sense, see the comment about the 512b one

Reply:
I don't know, the 512 bits is mentioned in the doc as a supported key size